nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Discovery.c

```diff
@@ -1073,7 +1073,9 @@ NFCSTATUS phNciNfc_UpdateDiscConfigParams(void *pNciHandle,
     uint8_t bIndex=1;
     PH_LOG_NCI_FUNC_ENTRY();
 
-    if(pPollConfig->PollNfcAActive)
+    if(pPollConfig->PollNfcAActive ||
+        ((pPollConfig->PollNfcFActive && pNciContext->ResetInfo.NciVer & PH_NCINFC_VERSION_MAJOR_MASK) ==
+         (PH_NCI2xNFC_VERSION & PH_NCINFC_VERSION_MAJOR_MASK)))
```

Comment:
A comment here would really help to indicate NFC A or NFC F v2 are being checked.